listenbrainz/webserver/views/feedback_api.py

```diff
@@ -0,0 +1,125 @@
+import ujson
+import listenbrainz.db.user as db_user
+import listenbrainz.db.feedback as db_feedback
+
+from datetime import datetime
+from flask import Blueprint, current_app, jsonify, request
+from listenbrainz.webserver.decorators import crossdomain
+from listenbrainz.webserver.errors import (APIBadRequest,
+                                           APIInternalServerError, APINotFound,
+                                           APIServiceUnavailable,
+                                           APIUnauthorized)
+from listenbrainz.webserver.rate_limiter import ratelimit
+from listenbrainz.webserver.views.api import _validate_auth_header, _parse_int_arg
+from listenbrainz.webserver.views.api_tools import log_raise_400, is_valid_uuid
+from listenbrainz.feedback import Feedback
+from pydantic import ValidationError
+
+feedback_api_bp = Blueprint('feedback_api_v1', __name__)
+
+
+@feedback_api_bp.route("submit-feedback", methods=["POST", "OPTIONS"])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def submit_feedback():
+    """
+    Submit recording feedback (love/hate) to the server. A user token (found on  https://listenbrainz.org/profile/ )
+    must be provided in the Authorization header! Each request should contain only one feedback in the payload.
+
+    For complete details on the format of the JSON to be POSTed to this endpoint, see :ref:`feedback-json-doc`.
+
+    :reqheader Authorization: Token <user token>
+    :statuscode 200: feedback accepted.
+    :statuscode 400: invalid JSON sent, see error message for details.
+    :statuscode 401: invalid authorization. See error message for details.
+    :resheader Content-Type: *application/json*
+    """
+    user = _validate_auth_header()
+
+    data = request.json
+
+    if 'recording_msid' not in data or 'score' not in data:
+        log_raise_400("JSON document must contain recording_msid and "
+                      "score top level keys", data)
+
+    if 'recording_msid' in data and 'score' in data and len(data) > 2:
+        log_raise_400("JSON document may only contain recording_msid and "
+                      "score top level keys", data)
+
+    try:
+        feedback = Feedback(user_id=user["id"], recording_msid=data["recording_msid"], score=data["score"])
+    except ValidationError as e:
+        log_raise_400("Invalid JSON document submitted: %s" % str(e).replace("\n ", ":").replace("\n", " "),
+                      data)  # str.replace() to tidy up the error message
```

Comment:
this seems random, do we really need to do this? If yes, we should have a more detailed comment saying what this does.

Reply:
Improved the comment.